src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py

```diff
@@ -1315,3 +1315,72 @@ def test_sql_db_import_export_mgmt(self, resource_group, resource_group_location
                      JMESPathCheck('resourceGroup', resource_group),
                      JMESPathCheck('serverName', server),
                      JMESPathCheck('status', 'Completed')])
+
+
+class SqlServerVnetMgmtScenarioTest(ScenarioTest):
+
+    @ResourceGroupPreparer(location='eastus2euap')
+    @SqlServerPreparer(location='eastus2euap')
+    def test_sql_vnet_mgmt(self, resource_group, resource_group_location, server):
+        rg = resource_group
+        vnet_rule_1 = 'rule1'
+
+        # Create vnet's - vnet1 and vnet2
+
+        vnetName1 = 'vnet1'
+        vnetName2 = 'vnet2'
+        subnetName = 'subnet1'
+        addressPrefix = '10.0.1.0/24'
+        endpoint = 'Microsoft.Sql'
+
+        # Vnet 1
+        self.cmd('network vnet create -g {} -n {}'.format(rg, vnetName1))
+        self.cmd('network vnet subnet create -g {} --vnet-name {} -n {} --address-prefix {} --service-endpoints {}'
+                 .format(rg, vnetName1, subnetName, addressPrefix, endpoint),
+                 checks=JMESPathCheck('serviceEndpoints[0].service', 'Microsoft.Sql'))
+
+        vnet1 = self.cmd('network vnet subnet show -n {} --vnet-name {} -g {}'
+                         .format(subnetName, vnetName1, rg)).get_output_in_json()
+        vnet_id_1 = vnet1['id']
+
+        # Vnet 2
+        self.cmd('network vnet create -g {} -n {}'.format(rg, vnetName2))
+        self.cmd('network vnet subnet create -g {} --vnet-name {} -n {} --address-prefix {} --service-endpoints {}'
+                 .format(rg, vnetName2, subnetName, addressPrefix, endpoint),
+                 checks=JMESPathCheck('serviceEndpoints[0].service', 'Microsoft.Sql'))
+
+        vnet2 = self.cmd('network vnet subnet show -n {} --vnet-name {} -g {}'
+                         .format(subnetName, vnetName2, rg)).get_output_in_json()
+        vnet_id_2 = vnet2['id']
+
+        # test sql server vnet-rule create
+        self.cmd('sql server vnet-rule create --name {} -g {} --server {} --subnet {}'
```

Comment:
sorry, I still do not see a test that creates vnet-rule with --vnet-name param.

Reply:
ah! I thought you meant testing it locally. I can add another test for that.